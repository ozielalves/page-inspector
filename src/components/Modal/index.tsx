import React, { ReactNode } from "react";
import styled from "styled-components";
import { SubTitle, Text, Wrapper } from "../../styles";
import CircularProgress from "../CircularProgress";
import { ReactComponent as CloseIcon } from "../../assets/close-icon.svg";
import ActionButton from "../ActionButton";

interface RequestDetailsProps {
  open: boolean;
  onClose: () => void;
  children?: ReactNode;
  title?: string;
}

const RequestDetails = ({
  open,
  onClose,
  children,
  title,
}: RequestDetailsProps) => {
  return open ? (
    <ModalBackdrop
      onClick={() => {
        onClose();
      }}
    >
      <ModalWrapper
        onClick={(e) => {
          e.stopPropagation();
        }}
      >
        <ModalHeader>
          <SubTitle>{title}</SubTitle>
          <ActionButton
            onClick={() => {
              onClose();
            }}
          >
            <CloseIcon />
          </ActionButton>
        </ModalHeader>
        <ModalContent>{children}</ModalContent>
      </ModalWrapper>
    </ModalBackdrop>
  ) : null;
};

const ModalHeader = styled.div`
  width: 100%;
  display: flex;
  justify-content: center;
  align-items: center;
  position: relative;
  padding-bottom: 10px;
  border: 1px solid var(--base-color-border);
  border-left: none;
  border-right: none;
  border-top: none;
`;

const ModalBackdrop = styled.div`
  position: fixed;
  top: 0;
  bottom: 0;
  left: 0;
  right: 0;
  background-color: rgba(0, 0, 0, 0.4);
  display: flex;
  justify-content: center;
  align-items: center;
  z-index: 2;
`;

const ModalContent = styled.div`
  width: 100%;
  height: 100%;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
`;

const ModalWrapper = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: flex-start;
  align-items: center;
  height: 70%;
  min-height: 200px;
  width: 60%;
  background-color: var(--base-color-white);
  box-shadow: var(--base-color-shadow) 0px 4px 10px 0px;
  border-radius: 50px;
  padding: 37px 65px;

  ._button {
    position: absolute;
    right: 0;
  }

  button {
    padding: 0;
    height: 40px;
    width: 40px;
  }

  @media (max-width: 600px) {
    width: 80%;
    padding: 30px;
    max-height: 60%;
  }

  @media (max-width: 380px) {
    width: 90%;
    padding: 30px;
    max-height: 70%;
  }
`;

export default RequestDetails;
