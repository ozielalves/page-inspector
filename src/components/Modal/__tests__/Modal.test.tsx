import { shallow } from "enzyme";
import renderer from "react-test-renderer";
import React from "react";
import Modal from "../index";

import ActionButtom from "../../ActionButton";

describe("Modal render test", () => {
  it("renders correctly ActionButtom", () => {
    const wrapper = shallow(
      <Modal onClose={() => {}} open={true}/>
    );
    expect(wrapper.find(ActionButtom)).toBeTruthy();
  });

  it("should render the correct Modal title", () => {
    const tree = renderer
      .create(<Modal onClose={() => {}} open={true} title="Normal Title" />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
