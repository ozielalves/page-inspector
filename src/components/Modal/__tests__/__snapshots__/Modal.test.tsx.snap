// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal render test should render the correct Modal title 1`] = `
<div
  className="sc-crrsfI hdGniU"
  onClick={[Function]}
>
  <div
    className="sc-bqyKva forjPo"
    onClick={[Function]}
  >
    <div
      className="sc-iqHYGH bGFLjY"
    >
      <p
        className="sc-gsTCUz sc-dlfnbm hFbaQT kWkuPj"
      >
        Normal Title
      </p>
      <div
        className="sc-kEjbxe jqePyN"
        style={
          Object {
            "position": "absolute",
            "right": 0,
          }
        }
      >
        <button
          onClick={[Function]}
          style={
            Object {
              "background": "transparent",
            }
          }
        >
          <svg>
            close-icon.svg
          </svg>
        </button>
      </div>
    </div>
    <div
      className="sc-dQppl iASszO"
    />
  </div>
</div>
`;
